_includes/assets/js/indicatorModelHelpers.js

```diff
@@ -0,0 +1,539 @@
+        return [];
+      }
+
+      return data[keys[0]].map(function(item, index) {
+        return _.object(keys, keys.map(function(key) {
+          return data[key][index];
+        }));
+      });
+    },
+    extractUnique: function(prop, data) {
+      return _.chain(data).pluck(prop).uniq().filter(function(f) { return f; }).sortBy(function(year) {
+        return year;
+      }).value();
+    },
+    getColumnsFromData: function(data) {
+      return Object.keys(data[0]);
+    },
+    dataHasColumn: function(column, data) {
+      return this.getColumnsFromData(data).includes(column);
+    },
+    dataHasUnits: function(data) {
+      return this.dataHasColumn(this.UNIT_COLUMN, data);
+    },
+    dataHasGeoCodes: function(data) {
+      return this.dataHasColumn(this.GEOCODE_COLUMN, data);
+    },
+    getFirstUnitInData: function(data) {
+      return data.find(function(row) {
+        return row[this.UNIT_COLUMN];
+      }, this)[this.UNIT_COLUMN];
+    },
+    getDataByUnit: function(data, unit) {
+      if (!unit) {
+        return data;
+      }
+      return data.filter(function(row) {
+        return row[this.UNIT_COLUMN] === unit;
+      }, this);
+    },
+    getDataBySelectedFields: function(data, selectedFields) {
+      return data.filter(function(row) {
+        return selectedFields.some(function(field) {
+          return field.values.includes(row[field.field]);
+        });
+      });
+    },
+    getDataMatchingCombination: function(data, combination, selectableFields) {
+      return data.filter(function(row) {
+        return selectableFields.every(function(field) {
+          return row[field] === combination[field];
+        });
+      });
+    },
+    getUnitFromStartValues: function(startValues) {
+      var match = startValues.find(function(startValue) {
+        return startValue.field === this.UNIT_COLUMN;
+      }, this);
+      return (match) ? match.value : false;
+    },
+    selectFieldsFromStartValues: function(startValues, selectableFields) {
+      if (!startValues) {
+        return [];
+      }
+      var specialFields = this.nonFieldColumns();
+      var allowedStartValues = startValues.filter(function(startValue) {
+        var normalField = !specialFields.includes(startValue.field);
+        var allowedField = selectableFields.includes(startValue.field)
+        return normalField && allowedField;
+      });
+      var valuesByField = {};
+      allowedStartValues.forEach(function(startValue) {
+        if (!(startValue.field in valuesByField)) {
+          valuesByField[startValue.field] = [];
+        }
+        valuesByField[startValue.field].push(startValue.value);
+      });
+      return Object.keys(valuesByField).map(function(field) {
+        return {
+          field: field,
+          values: valuesByField[field],
+        };
+      });
+    },
+    selectMinimumStartingFields: function(data, fieldNames) {
+      // We filter our full dataset to only those fields.
```

Comment:
Give priority in making your code readable instead of writing comments to explain it.